packages/http-framework/src/lib/Client.ts

```diff
@@ -19,13 +28,53 @@ container.stores.register(new CommandStore());
 container.stores.register(new InteractionHandlerStore());
 container.stores.register(new ListenerStore());
 
-export class Client extends EventEmitter {
+export interface ClientEventCommandContext {
+	command: Command;
+	interaction: APIApplicationCommandInteraction;
+	response: ServerResponse;
+}
+
+export interface ClientEventAutocompleteContext {
+	command: Command;
+	interaction: APIApplicationCommandAutocompleteInteraction;
+	response: ServerResponse;
+}
+
+export interface ClientEventInteractionHandlerContext {
+	handler: InteractionHandler;
+	interaction: APIMessageComponentInteraction | APIModalSubmitInteraction;
+	response: ServerResponse;
+}
+
+export interface ClientEvents {
+	commandNameMissing: [interaction: APIApplicationCommandAutocompleteInteraction, response: ServerResponse];
+	commandNameUnknown: [interaction: APIApplicationCommandInteraction | APIApplicationCommandAutocompleteInteraction, response: ServerResponse];
+	commandMethodUnknown: [context: ClientEventCommandContext];
+	commandRun: [context: ClientEventCommandContext];
+	commandSuccess: [context: ClientEventCommandContext, value: unknown];
+	commandError: [error: unknown, context: ClientEventCommandContext];
+	commandFinish: [context: ClientEventCommandContext];
+	autocompleteRun: [context: ClientEventAutocompleteContext];
+	autocompleteSuccess: [context: ClientEventAutocompleteContext, value: unknown];
+	autocompleteError: [error: unknown, context: ClientEventAutocompleteContext];
+	autocompleteFinish: [context: ClientEventAutocompleteContext];
+	interactionHandlerNameInvalid: [interaction: APIMessageComponentInteraction | APIModalSubmitInteraction, response: ServerResponse];
+	interactionHandlerNameUnknown: [interaction: APIMessageComponentInteraction | APIModalSubmitInteraction, response: ServerResponse];
+	interactionHandlerRun: [context: ClientEventInteractionHandlerContext];
+	interactionHandlerSuccess: [context: ClientEventInteractionHandlerContext, value: unknown];
+	interactionHandlerError: [error: unknown, context: ClientEventInteractionHandlerContext];
+	interactionHandlerFinish: [context: ClientEventInteractionHandlerContext];
+}
+
+type MappedClientEvents = { [K in keyof ClientEvents]: ClientEvents[K] };
```

Comment:
Move this to a separate file `ClientEvents.ts` to keep the LoCs of this file lower.